source/Nevermore/ParameterNameGenerator.cs

```diff
@@ -1,16 +1,17 @@
 ﻿namespace Nevermore
 {
-    public interface IParameterNameGenerator
+    public interface IUniqueParameterGenerator
     {
-        string GenerateUniqueParameterName(string parameterDescription);
+        UniqueParameter GenerateUniqueParameterName(Parameter parameter);
     }
 
-    class ParameterNameGenerator : IParameterNameGenerator
+    class UniqueParameterGenerator : IUniqueParameterGenerator
```

Comment:
If you wanted to get really protective, you can restrict the UniqueParameter class so it is only able to be created inside the UniqueParameterGenerator.  https://stackoverflow.com/questions/1664793/how-to-restrict-access-to-nested-class-member-to-enclosing-class.  See my PR here: https://github.com/OctopusDeploy/Nevermore/pull/46